tests/upstream_switch/case.sh

```diff
@@ -217,6 +227,47 @@ function test() {
 	gen_incr_data
 	verify_result
 	clean_task
+	echo "CASE=test_relay success"
+}
+
+function test_binlog_filename_change_when_enable_gtid() {
+	cleanup_process
+	check_master
+	clean_data
+	setup_replica
+	gen_full_data
+	run_dm_components_and_create_sources
+	start_task
+
+	# flush mysql57 master binlog files -> mysql-bin.000002
+	exec_sql $host1 "flush logs;"
+	# flush mysql8 master binlog files -> mysql-bin.000004
+	exec_sql $host2 "flush logs;"
+	exec_sql $host2 "flush logs;"
+	exec_sql $host2 "flush logs;"
+
+	# make mysql8 master slave switch
+	docker-compose -f $CUR/docker-compose.yml pause mysql8_master
+	wait_mysql $host2 2
+
+	# make a flush job to dm
+	exec_sql $host1 "alter table ${db}.${tb} add column c int;"
+	exec_sql $host2 "alter table ${db}.${tb} add column c int;"
+
+	check_sync_diff $WORK_DIR $CUR/conf/diff_config.toml
+	# mysql8 new master's binlog file still is mysql-bin.000001
```

Comment:
why we flush mysql 5.7, did you want to test it but forget?

Reply:
i want last query-status result source-1's binlog filnname is not `binlog.00001` so that i can check and count source-2's binlog filename

Reply:
the number of `binlog.00001` will be 4 not 2 then